common/lib/driver-definitions/src/storage.ts

```diff
@@ -295,12 +295,27 @@ export interface IDocumentService {
     dispose(error?: any): void;
 }
 
+/**
+ * Interface document service describing capabilities
+ */
+export interface IDocumentServiceCapabilities{
+    /**
+     * Document service can create docs with non-zero seq. start #
+     */
+    acceptsNonZeroSeqStartNum: boolean,
+}
+
 export interface IDocumentServiceFactory {
     /**
      * Name of the protocol used by factory
      */
     protocolName: string;
 
+    /**
+     * Surface document service capabilities
+     */
+    capabilities?: IDocumentServiceCapabilities;
```

Comment:
I don't think this is the right place to add such capabilities. Couple of reasons why:
1. It is adding a summary / snapshot specific feature and if so, it belongs to IDocumentStorageService which deals with snapshot upload / download.
2. The container shouldn't care about this. It should work for non-zero seq number and let the driver layer handle this.

I understand that this is needed in `rehydrateDetachedFromSnapshot` in container.ts and we don't have access to the IDocumentStorageService at that time. So, I can see couple of ways to make this work without changing the loader / driver contract:
1. Make it similar to the "attachment blobs for detached container" feature. It fails when trying to attach the container. It also looks at the loader services [here](https://github.com/microsoft/FluidFramework/blob/45f6262032d33ad839dc0133deccd799231bb854/packages/loader/container-loader/src/container.ts#L799) to see if the container has blobs in detached mode. This means that the user will only see the failure when attaching the container and can keep working on it before that. I think this is fine and should be no different than when attach fails for other reasons such as invalid token, invalid permissions, etc.
2. There should be a service that provides this functionality of downloading a previous summary and rehydrating the container with that. It can validate that this is done only for FRS and only then call the container.

Reply:
On your first two points above:
1. `IDocumentStorageService` is folded under `IDocumentService`, which seems to represent both storage and delta services capabilities. Unfortunately, we don't have access to both until container is attached. Exposing `capabilities` seemed not too different then exposing `policies` like `storageOnly`. Yes - the difference here is that this particular policy is around connection/creation.
2. We chatted about this briefly, Unless we fail fast here, user can rehydrate container, make changes, only to realize later there is error on attach. 

On the suggestions you had:
1. This case is interesting, but it seems like this may be (a) API usage error or (b) these sort of errors could have been caught earlier? I'm not sure why this pattern is better than to safeguard and fail early.
2. Like another `ILoaderServices` ? This seems like a lot of overhead and functionality duplication for something that may go away when we achieve parity with ODSP.

@vladsud could you give us your take here?

Reply:
I'd say it's not worth complicating interfaces - we should just say that every service supports creation of files with non-zero sequence numbers. If it's not supported driver throws an error. And if we need to stage it, we simply document that certain storage does not support certain workflow.

Reply:
Ok. I have moved the check to `OdspDocumentServiceFactoryCore`. It will fail with "Seq number in detached ODSP container should be 0" general error. Container loader is not checking for non-zero sequence numbers any further.